### PR TITLE
Hubspot - New Sources and Updates

### LINKED ISSUE #5593: [FEATURE] Hubspot - Deal Updated & New Deal: Add Deal Pipeline prop and Deal Stage prop for filtering events
**Is your feature request related to a problem? Please describe.**
For Hubspot Deal Updated and New Deal source, currently all events will be emitted as there's no props for filtering.
image
image

**Describe the solution you'd like**
A customer would like to have `Deal Pipeline` and `Deal Stage` prop to only emit event for the Deals in the specific Deal Pipeline and Deal Stage

**Do you have a workaround?**
We can use Pipedream Filter helpers

**Comparable features in other tools?**
Zapier has those 2 props for filtering events
image

**Additional context**
First reported here: https://pipedream-users.slack.com/archives/G01J50YGJGP/p1677708767986359

### LINKED ISSUE #5594: [TRIGGER] New Hubspot sources
**Describe the event source. What app is this for, and what event does the trigger correspond to?**
- Company Property Change
- New Deal Property Change
- New Ticket Property Change
- New Contact Property Change

Zapier offers the following sources here
image

**Please provide a link to the relevant API docs for the specific service / operation this trigger is tied to.**
I'm not sure about the API Docs. Please check if the [Hubspot API Client](https://www.npmjs.com/package/@hubspot/api-client) if they provide enough capabilities for us 🙏

### PR DESCRIPTION
**Pull request by michelle0927**
Resolves #5593 

**Updates Sources**

- New Deal
- Deal Updated

Resolves #5594 

**New Sources**

- New Deal Property Change
- New Ticket Property Change
- New Company Property Change
- New Contact Property Change

### CODE REVIEW

File: components/hubspot/sources/deal-updated/deal-updated.mjs [hunk 1]
```diff
@@ -5,9 +5,33 @@ export default {
   key: "hubspot-deal-updated",
   name: "Deal Updated",
   description: "Emit new event each time a deal is updated.",
```
Comment (alysonturing) [inline components/hubspot/sources/deal-updated/deal-updated.mjs:7]: is it possible to add the link to the docs?

File: components/hubspot/sources/new-company-property-change/new-company-property-change.mjs [hunk 2]
```diff
@@ -0,0 +1,77 @@
+import common from "../common.mjs";
+
+export default {
+  ...common,
+  key: "hubspot-new-company-property-change",
+  name: "New Company Property Change",
+  description: "Emit new event when a specified property is provided or updated on a company.",
```
Comment (alysonturing) [inline components/hubspot/sources/new-company-property-change/new-company-property-change.mjs:7]: is it possible to add the link to the docs?

File: components/hubspot/sources/new-contact-property-change/new-contact-property-change.mjs [hunk 3]
```diff
@@ -0,0 +1,77 @@
+import common from "../common.mjs";
+
+export default {
+  ...common,
+  key: "hubspot-new-contact-property-change",
+  name: "New Contact Property Change",
+  description: "Emit new event when a specified property is provided or updated on a contact.",
```
Comment (alysonturing) [inline components/hubspot/sources/new-contact-property-change/new-contact-property-change.mjs:7]: is it possible to add the link to the docs?

File: components/hubspot/sources/deal-updated/deal-updated.mjs [hunk 4]
```diff
@@ -5,9 +5,33 @@ export default {
   key: "hubspot-deal-updated",
   name: "Deal Updated",
```
Comment (alysonturing) [inline components/hubspot/sources/deal-updated/deal-updated.mjs:6]: Source names should start with "New".

File: components/hubspot/sources/deal-updated/deal-updated.mjs [hunk 5]
```diff
@@ -5,9 +5,33 @@ export default {
...
+          pipeline: c.pipeline,
+        }),
+      ],
+      type: "string",
+      label: "Stage",
+      description: "Filter deals by stage",
+      optional: true,
+    },
+  },
   hooks: {},
```
Comment (alysonturing) [inline components/hubspot/sources/deal-updated/deal-updated.mjs:35]: do we need an empty hooks property?

File: components/hubspot/sources/new-company-property-change/new-company-property-change.mjs [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+      type: "string",
+      label: "Property",
+      description: "The company property to watch for changes",
+      async options() {
+        const { results: properties } = await this.hubspot.getProperties("companies");
+        return properties.map((property) => property.name);
+      },
+    },
+  },
+  hooks: {},
```
Comment (alysonturing) [inline components/hubspot/sources/new-company-property-change/new-company-property-change.mjs:23]: empty hooks section

File: components/hubspot/sources/new-contact-property-change/new-contact-property-change.mjs [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+      type: "string",
+      label: "Property",
+      description: "The contact property to watch for changes",
+      async options() {
+        const properties = await this.hubspot.getContactProperties();
+        return properties.map((property) => property.name);
+      },
+    },
+  },
+  hooks: {},
```
Comment (alysonturing) [inline components/hubspot/sources/new-contact-property-change/new-contact-property-change.mjs:23]: empty hooks section

File: components/hubspot/sources/new-deal-property-change/new-deal-property-change.mjs [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+      type: "string",
+      label: "Property",
+      description: "The deal property to watch for changes",
+      async options() {
+        const properties = await this.hubspot.getDealProperties();
+        return properties.map((property) => property.name);
+      },
+    },
+  },
+  hooks: {},
```
Comment (alysonturing) [inline components/hubspot/sources/new-deal-property-change/new-deal-property-change.mjs:23]: empty hooks section

File: components/hubspot/sources/new-deal-property-change/new-deal-property-change.mjs [hunk 9]
```diff
@@ -0,0 +1,77 @@
+import common from "../common.mjs";
+
+export default {
+  ...common,
+  key: "hubspot-new-deal-property-change",
+  name: "New Deal Property Change",
+  description: "Emit new event when a specified property is provided or updated on a deal.",
```
Comment (alysonturing) [inline components/hubspot/sources/new-deal-property-change/new-deal-property-change.mjs:7]: is it possible to add the link to the docs?

File: components/hubspot/sources/new-deal/new-deal.mjs [hunk 10]
```diff
@@ -5,9 +5,33 @@ export default {
   key: "hubspot-new-deal",
   name: "New Deals",
   description: "Emit new event for each new deal created.",
```
Comment (alysonturing) [inline components/hubspot/sources/new-deal/new-deal.mjs:7]: is it possible to add the link to the docs?

File: components/hubspot/sources/new-deal/new-deal.mjs [hunk 11]
```diff
@@ -5,9 +5,33 @@ export default {
...
+          pipeline: c.pipeline,
+        }),
+      ],
+      type: "string",
+      label: "Stage",
+      description: "Filter deals by stage",
+      optional: true,
+    },
+  },
   hooks: {},
```
Comment (alysonturing) [inline components/hubspot/sources/new-deal/new-deal.mjs:35]: empty hooks section

File: components/hubspot/sources/new-ticket-property-change/new-ticket-property-change.mjs [hunk 12]
```diff
@@ -0,0 +1,77 @@
+import common from "../common.mjs";
+
+export default {
+  ...common,
+  key: "hubspot-new-ticket-property-change",
+  name: "New Ticket Property Change",
+  description: "Emit new event when a specified property is provided or updated on a ticket.",
```
Comment (alysonturing) [inline components/hubspot/sources/new-ticket-property-change/new-ticket-property-change.mjs:7]: is it possible to add the link to the docs?

File: components/hubspot/sources/new-ticket-property-change/new-ticket-property-change.mjs [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+      type: "string",
+      label: "Property",
+      description: "The ticket property to watch for changes",
+      async options() {
+        const { results: properties } = await this.hubspot.getProperties("tickets");
+        return properties.map((property) => property.name);
+      },
+    },
+  },
+  hooks: {},
```
Comment (alysonturing) [inline components/hubspot/sources/new-ticket-property-change/new-ticket-property-change.mjs:23]: empty hooks section

**Review by alysonturing — APPROVED** [review]
Thanks for your contribution, it was approved, but added some comments, please check if it's relevant and change accordingly if needed

**Comment by vunguyenhung** [comment]
Hello everyone, I have tested this PR and there're some test cases failed or needed improvement. 

Please check the test report below for more information
[Hubspot_sources_5605_1106.pdf](https://github.com/PipedreamHQ/pipedream/files/10878081/Hubspot_sources_5605_1106.pdf)

File: components/hubspot/sources/deal-updated/deal-updated.mjs (hunk 5 above)
Comment (michelle0927) [reply components/hubspot/sources/deal-updated/deal-updated.mjs]: This is to overwrite the hooks section in `common.mjs`.

**Comment by vunguyenhung** [comment]
Hi everyone, all test cases are passed! Ready for release! 

Test report
[Hubspot_sources_5605_0753.pdf](https://github.com/PipedreamHQ/pipedream/files/10893443/Hubspot_sources_5605_0753.pdf)

**Review by GTFalcao — APPROVED** [review]

**Pull request merged**